tests/unittests/test_handler/test_handler_apt_configure_sources_list_v3.py

```diff
@@ -108,37 +108,29 @@ def setUp(self):
         get_arch.return_value = 'amd64'
         self.addCleanup(apatcher.stop)
 
-    def _get_cloud(self, distro, metadata=None):
-        self.patchUtils(self.new_root)
-        paths = helpers.Paths({})
-        cls = distros.fetch(distro)
-        mydist = cls(distro, {}, paths)
-        myds = DataSourceNone.DataSourceNone({}, mydist, paths)
-        if metadata:
-            myds.metadata.update(metadata)
-        return cloud.Cloud(myds, paths, {}, mydist, None)
-
     def _apt_source_list(self, distro, cfg, cfg_on_empty=False):
         """_apt_source_list - Test rendering from template (generic)"""
         # entry at top level now, wrap in 'apt' key
         cfg = {'apt': cfg}
-        mycloud = self._get_cloud(distro)
-
-        with mock.patch.object(util, 'write_file') as mock_writefile:
-            with mock.patch.object(util, 'load_file',
-                                   return_value=MOCKED_APT_SRC_LIST
-                                   ) as mock_loadfile:
-                with mock.patch.object(os.path, 'isfile',
-                                       return_value=True) as mock_isfile:
-                    cfg_func = ('cloudinit.config.cc_apt_configure.' +
-                                '_should_configure_on_empty_apt')
-                    with mock.patch(cfg_func,
-                                    return_value=(cfg_on_empty, "test")
-                                    ) as mock_shouldcfg:
-                        cc_apt_configure.handle("test", cfg, mycloud, LOG,
-                                                None)
-
-        return mock_writefile, mock_loadfile, mock_isfile, mock_shouldcfg
+        mycloud = get_cloud(distro)
+
+        with ExitStack() as stack:
```

Comment:
The `get_cloud` call in these tests included a call to `patchUtils`, so updated this section to also mock out `del_file`. The rest is just syntactic sugar to not have so much nesting with the mocks.